kubo-rpc/src/lib.rs

```diff
@@ -288,6 +295,14 @@ impl IpfsDep for Arc<IpfsService> {
         &self,
         topic: String,
     ) -> Result<BoxStream<'static, anyhow::Result<GossipsubEvent>>, Error> {
+        {
+            self.topics
+                .lock()
+                .expect("should be able to lock topics set")
```

Comment:
Could this expect cause a panic? If so, is that ok?

Reply:
In theory it can cause a panic, but not in practice. From the docs:

> This function might panic when called if the lock is already held by the current thread.

We do not hold the lock and then call any code that would try to acquire the lock again. That is why we explicitly use `expect` in order to make it clear we do not expect the panic to ever occur in practice.